### ISSUE TITLE
Project status

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/awesome#1363]
I got pretty burned out by this project and haven't been active for many months. This project is called Awesome and lists are expected to be of (awesomely) high-quality. Unfortunately, many submissions were not and required a lot of effort on my part to just comment on things that were already listed in the pull request guidelines. This doesn't scale to one person when there are many new submissions a day.

I'm ready to put some work into this project again, but I need your help.

There's a new requirement for submissions now. When you submit a new list, you're required to review at least 2 other open pull requests. This is my experiment to try to make the Awesome project self-sustaining. [Read more here.](https://github.com/sindresorhus/awesome/blob/master/pull_request_template.md#requirements-for-your-pull-request)

**This requirement also applies to existing submissions, so if you like to see your submission merged, review at least 2 other pull requests and I'll prioritize getting your submission merged.**

Please also go through the [list guidelines](https://github.com/sindresorhus/awesome/blob/master/pull_request_template.md) again on your own submissions. You have probably missed something.

**Comment by sellisd** [comment]
Glad to see you back active on this! How about trying to automate part of the review process? At least some of the requirements for the pull request could be validated by a script I will put down some ideas here: https://github.com/sellisd/awesomeReview  to see if it can be of use

**Comment by transitive-bullshit** [comment]
@sellisd see [awesome-lint](https://github.com/sindresorhus/awesome-lint) which seems like a solid starting point, though it's in need of a lot of love.

**Comment by dspinellis** [comment]
Making the project self-sustaining and avoiding burnout is a good idea.  It would be helpful to provide a bit more guidance regarding the review process.  
* A list of PRs requiring review. Unless I'm missing something, now reviewed and unreviewed PRs all appear together.
* A way to mark PRs whose owners have fulfilled the "review at least two PRs" requirement or to notify you that the requirement has been fulfilled.

**Comment by TheKitty** [comment]
Reviewed 
https://github.com/TheKitty/awesome-microbit/pull/1
https://github.com/TheKitty/awesome-jupyter/pull/1

for 1160 and 1127

**Comment by troxler** [comment]
It seems obvious that the current approach of merging pull requests does not scale. Currently, there are [four pull requests that have been open for more than one year](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen+sort%3Acreated-asc) and there is a total of 102 open pull requests. At least some of them seem to have met all requirements including the authors having done the reviews of other pull requests. Judging from [the list of commits](https://github.com/sindresorhus/awesome/commits/master), @sindresorhus is the only (active) user with merge privilege.

I have seen some pull requests that were closed by their respective authors because they lost patience or interest. The process of being listed is really cumbersome and I understand everybody who is not willing to create a pull request just to have it open for months (or even over a year) without any reasonable prospect of being listed at all.

I understand that maintaining this project is time-consuming and often even annoying. But letting this project die a slow death is a pity. This only works when more people with proper privileges are involved. Sindre, I think it is time to add more contributers to this repository. What do you think?

**Comment by sindresorhus** [comment]
@troxler Just because there are lots of open PRs doesn't mean this project is dying. This repo is just a list of what I consider good. The beauty of open source is that anyone can create a list. It doesn't have to be listed here. There are currently 1500 repos tagged with the [`awesome-list`](https://github.com/topics/awesome-list) tag and probably lots more that are untagged. I try to prioritize reviewing PRs here, but most PR reviews are still just me copy-pasting items from the contribution guidelines, which is very demotivating. I also have a lot of other OSS responsibilities.

I have tried adding more collaborators multiple times before without success. You don't even need to be a collaborator to help out. Merging the PRs are easy. Thoroughly reviewing them is the time consuming part, and anyone can help out with that, but few do.

I'm also working with community members on a [awesome linter](https://github.com/sindresorhus/awesome-lint) which will hopefully lessen my PR review work.

**Comment by troxler** [comment]
> Just because there are lots of open PRs doesn't mean this project is dying.

No it does not indeed. I've been following this project for a while and I think during this time, the number of open pull requests even decreased. But many of them were closed because their authors apparently lost interest and gave up or just did not answer anymore. I strongly think that is also has to do with the amount of time it takes from the request creation to actually being merged (or closed).

> I try to prioritize reviewing PRs here, but most PR reviews are still just me copy-pasting items from the contribution guidelines, which is very demotivating. I also have a lot of other OSS responsibilities.

I do realize that (kudos and thanks btw!). And I especially get the part "demotivating" as I've seen many examples of authors that clearly did not read the guidelines at all.

> You don't even need to be a collaborator to help out. Merging the PRs are easy. Thoroughly reviewing them is the time consuming part, and anyone can help out with that, but few do.

That may be true. But don't you need tools to see at a glance which pull requests could potentially be merged (i.e. someone reviewed it and thinks it is OK) and which need more work? Do you go through all 100+ requests regularly just to check that? Wouldn't it be helpful to have appropriate labels or a being able to [filter by approvals](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen+review%3Aapproved)? Contributors without privileges can't use either. It would also make sense for the reviewers to quickly find requests that actually need a review without skimming through a long list first. And personally, I also find it demotivating to do reviews without proper tools.

**Comment by sindresorhus** [comment]
> I strongly think that is also has to do with the amount of time it takes from the request creation to actually being merged (or closed).

Yup, but in most cases it takes a while because people keep opening PRs without having put much effort into their list or PR, so they require a lot of review time, which means I have less time for other PRs, and suddenly there are 100+ open PRs...

> Do you go through all 100+ requests regularly just to check that

I go through my Awesome repo notifications once in awhile to see if any new PRs are ready for another review.

> Wouldn't it be helpful to have appropriate labels or a being able to filter by approvals? Contributors without privileges can't use either.

Sure, this all sounds nice, but you don't just add random people to a project and hope they do a good job. You add people that already regularly do a good job.

> It would also make sense for the reviewers to quickly find requests that actually need a review without skimming through a long list first.

All PRs need review, including already reviewed ones. Even when I get external review help, they miss a lot of things, so the more reviews of the same PR, the better.

**Comment by troxler** [comment]
> Yup, but in most cases it takes a while because people keep opening PRs without having put much effort into their list or PR, so they require a lot of review time, which means I have less time for other PRs, and suddenly there are 100+ open PRs...

Yes I can imagine. First thing I would do is to close the PRs of authors that did respond to any comments in months. It seems useless to let people do yet another review on those anyway.

> All PRs need review, including already reviewed ones. Even when I get external review help, they miss a lot of things, so the more reviews of the same PR, the better.

That would be ideal yes. What currently is happening is the following: Some PRs (mostly newer ones) get multiple repetitive reviews while other PRs (mostly older ones) don't get any. My own PR has been open for exactly one year today and did not get a review. I tried pinging you, I updated the list several times to comply with updated rules, I ran the linter and did some more updates, I did reviews on other PRs and pinged you again. Well, a year has passed since I created the PR and it is still open. How would anyone be interested to go through all that hassle just to be ignored?

I highly appreciate what you do (here and on the many other projects you have) and I definitely understand that it all takes a lot of time. All I'm saying is that it does not seem to scale. Having that many open PRs of which many are so old does not attract new contributions in my opinion.

That is all I wanted to say :-)
